common/src/main/java/io/netty/util/AbstractConstant.java

```diff
@@ -27,15 +24,16 @@
 
     private final int id;
     private final String name;
-    private volatile long uniquifier;
-    private ByteBuffer directBuffer;
+    private final long uniquifier;
+    private static final AtomicLong uniqueIdGenerator = new AtomicLong();
```

Comment:
Is it possible that we may have may `AbstractConstant` objects allocated? If so consider using `AtomicLongFieldUpdater` instead of `AtomicLong`.